ScoutSuite/providers/gcp/resources/projects.py

```diff
@@ -0,0 +1,18 @@
+from ScoutSuite.providers.gcp.resources.resources import GCPCompositeResources
+
+class Projects(GCPCompositeResources):
+    def __init__(self, gcp_facade):
+        self.gcp_facade = gcp_facade
+
+    async def fetch_all(self, **kwargs):
+        raw_projects = await self.gcp_facade.get_projects()
+        self['projects'] = {}
+        for raw_project in raw_projects:
+            project_id = raw_project['projectId']
+            self['projects'][project_id] = {}
+            await self._fetch_children(self['projects'][project_id], gcp_facade = self.gcp_facade, project_id = project_id)
```

Comment:
I believe this should be parallelized. It would significantly improve performance. For AWS, it looks something like this:

```py
tasks = {
    asyncio.ensure_future(
        self._fetch_children(resource_parent=resource_parent, scope=scopes[resource_parent_key])
    ) for (resource_parent_key, resource_parent) in resources.items()
}
await asyncio.wait(tasks)
```

Reply:
Done!